docs/source/common/lightning_cli.rst

```diff
@@ -196,6 +194,74 @@ Note that there is a section for each class (model and trainer) including all th
+that might not be clear by only reading the help are the following.
+
+:class:`~pytorch_lightning.utilities.cli.LightningCLI` is based on argparse and as such follows the same arguments style
+as many POSIX command line tools. Long options are prefixed with two dashes and its corresponding values should be
+provided with an empty space or an equal sign, as :code:`--option value` or :code:`--option=value`. The same options can
+be given multiple times and ones given later (to the right) override the previous ones. If a class has an init parameter
+that is required, it also has to be given as :code:`--option` which makes things explicit and more readable instead of
+relying on positional arguments.
+
+When calling a CLI, all options can be provided using individual arguments. However, given the large amount of options
+that the CLIs have, it is recommended to use a combination of config files and individual arguments. Therefore, a common
+pattern could be a single config file and only a few individual arguments that override defaults or values in the
+config, for example:
+
+.. code-block:: bash
+
+    $ python trainer.py --config experiment_defaults.yaml --trainer.max_epochs 100
+
+Another common pattern could be having multiple config files, two examples being:
+
+.. code-block:: bash
+
+    $ python trainer.py --config config1.yaml --config config2.yaml [...]
+    $ python trainer.py --trainer trainer.yaml --model model.yaml --data data.yaml [...]
+
+When running experiments in clusters it could be desired to use a config which needs to be accessed from a remote
+location. :class:`~pytorch_lightning.utilities.cli.LightningCLI` comes with `fsspec
+<https://filesystem-spec.readthedocs.io/en/stable/>`_ support which allows reading from many types of remote file
+systems. One example is if you have installed the `gcsfs <https://gcsfs.readthedocs.io/en/stable/>`_ then a config could
+be stored in an S3 bucket and accessed as:
+
+.. code-block:: bash
+
+    $ python trainer.py --config s3://bucket/config.yaml [...]
+
+In some cases people might what to pass an entire config in an environment variable, which could also be used instead of
+a path to a file, for example:
+
+.. code-block:: bash
+
+    $ python trainer.py --trainer "$TRAINER_CONFIG" --model "$MODEL_CONFIG" [...]
+
+An alternative for environment variables could be to instantiate the CLI with :code:`env_parse=True`. In this case the
+the help shows the names of the environment variables for all options. A global config would be given in
+:code:`PL_CONFIG` and there wouldn't be a need to specify any command line argument.
+
+It is also possible to set a path to a config file of defaults. If the file exists it would be automatically loaded
+without having to specify any command line argument. Arguments given would override the values in the default config
+file. To load a defaults file :code:`my_cli_defaults.yaml` in current working directory would be implemented as:
```

Comment:
```suggestion
It is also possible to set a path to a config file of defaults. If the file exists it would be automatically loaded
without having to specify any command line argument. Arguments given would override the values in the default config
file. Loading a defaults file :code:`my_cli_defaults.yaml` in the current working directory would be implemented as:
```